googleapiclient/discovery.py

```diff
@@ -950,6 +951,14 @@ def _set_service_methods(self):
     self._add_next_methods(self._resourceDesc, self._schema)
 
   def _add_basic_methods(self, resourceDesc, rootDesc, schema):
+    # If this is the root Resource, add a new_batch_http_request() method.
+    batch_uri = '%s%s' % (
```

Comment:
Move this calculation inside your if since its result will only be used inside your if.